pkg/kubelet/kubelet.go

```diff
@@ -747,6 +750,8 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		*kubeCfg.MemoryThrottlingFactor,
 		kubeDeps.PodStartupLatencyTracker,
 		kubeDeps.TracerProvider,
+		tokenManager,
+		klet.kubeClient,
```

Comment:
kubeDeps.KubeClient will be nil for a kubelet not set up to talk to a kube-apiserver ... it looks like token.NewManager handles that case correctly if asked for a token:

```
			if c == nil {
				return nil, errors.New("cannot use TokenManager when kubelet is in standalone mode")
			}
```

does everything NewKubeGenericRuntimeManager does with the maybe-nil client we're passing down also handle a nil client safely?

Reply:
looks like it doesn't:
```
			getServiceAccount = func(namespace, serviceAccountName string) (*v1.ServiceAccount, error) {
				return kubeClient.CoreV1().ServiceAccounts(namespace).Get(ctx, serviceAccountName, metav1.GetOptions{})
			}
```

in general, passing a full clientset.Interface down into kubelet to use for ~anything that ~might be nil seems like opening a can of worms :-/

would it be smaller / safer to pass down a getServiceAccount() function to NewKubeGenericRuntimeManager instead of a full possibly-nil clientset?